src/NHibernate/Type/TimestampType.cs

```diff
@@ -1,142 +1,50 @@
 using System;
-using System.Collections;
-using System.Data.Common;
 using NHibernate.Engine;
-using NHibernate.SqlTypes;
 
 namespace NHibernate.Type
 {
 	/// <summary>
-	/// This is almost the exact same type as the DateTime except it can be used
-	/// in the version column, stores it to the accuracy the database supports, 
-	/// and will default to the value of DateTime.Now if the value is null.
```

Comment:
`default to the value of DateTime.Now if the value is null`: wrong, since it was "implemented" into `Set` which is not called when the value is `null`.
Moreover since when this comment was written, it seems `DateTimeType` has gained the ability to serve as version too.